docs/04-docker-setup-postgres-connect-with-r.md

```diff
@@ -37,50 +37,53 @@ system2("docker", "version", stdout = TRUE, stderr = TRUE)
+## Clean up if appropriate
+Remove the `cattle` and `sql-pet` containers if they exists (e.g., from a prior experiments).  
 
-chunk the following... 
+```r
+if (system2("docker", "ps -a", stdout = TRUE) %>% 
+   grepl(x = ., pattern = 'cattle') %>% 
+   any()) {
+     system2("docker", "rm -f cattle")
+}
+if (system2("docker", "ps -a", stdout = TRUE) %>% 
+   grepl(x = ., pattern = 'sql-pet') %>% 
+   any()) {
+     system2("docker", "rm -f sql-pet")
+}
+```
+
+The convention we use in this book is to assemble a command with `glue` so that the you can see all of its separate parts.  The following chunk just constructs the command, but does not execute it.  If you have problems executing a commnad, you can always copy the command and execute in your terminal session.
 
 ```r
 docker_cmd <- glue(
-  "run ",      # Run is the Docker command.  Everything that follows are `run` parameters.
+  "run ",      # Run is the Docker command.  Everything that follows are `docker run` parameters.
   "--detach ", # (or `-d`) tells Docker to disconnect from the terminal / program issuing the command
   "--name cattle ",       # tells Docker to give the container a name: `cattle`
   "--publish 5432:5432 ", # tells Docker to expose the Postgres port 5432 to the local network with 5432
   " postgres:10"  # tells Docker the image that is to be run (after downloading if necessary)
 )
-docker_cmd
-```
 
-```
-## run --detach --name cattle --publish 5432:5432  postgres:10
-```
-
-```r
-# Naming containers `cattle` for throw-aways and `pet` for ones we treasure and keep around.  :-)
+# We name containers `cattle` for throw-aways and `pet` for ones we treasure and keep around.  :-)
 ```
 
-Remove `cattle` if it exists.
-
 Submit the command constructed above:
 
 ```r
-system2("docker", docker_cmd, stdout = TRUE, stderr = TRUE)
+docker_cmd
 ```
 
 ```
-## Warning in system2("docker", docker_cmd, stdout = TRUE, stderr = TRUE):
-## running command ''docker' run --detach --name cattle --publish 5432:5432
-## postgres:10 2>&1' had status 125
+## run --detach --name cattle --publish 5432:5432  postgres:10
+```
+
+```r
+system2("docker", docker_cmd, stdout = TRUE, stderr = TRUE)
 ```
 
 ```
-## [1] "249b0658664a0d878ccfe6332878ee7f1e2ff8143706394db679793badd43e74"                                                                                                                                                                   
-## [2] "docker: Error response from daemon: driver failed programming external connectivity on endpoint cattle (a4f8447968a1778c9bb5a5d1a075f0842b6395b43e3d0d669a067c13f6227b5c): Bind for 0.0.0.0:5432 failed: port is already allocated."
-## attr(,"status")
-## [1] 125
+## [1] "596ad8bf2df8e1e21643dcd84b69de00e0d51ef3ef333fac685c74bc4ca0c32d"
```

Comment:
That long string of numbers is a hash code Docker created so you can refer to the container later. The code should capture this in a variable if there's any reason you want to refer to the specific container later on.

Reply:
I'm not sure when that string might be used, but I'm now capturing it.  Trapping one possible error.  We may find others as we go forward and there's now a place to check them.